### PR TITLE
Improve error handling

### PR DESCRIPTION
**Pull request by aliscott**
Show better error messages for:
- [x] Terraform binary missing
- [x] An unsupported Terraform version is used
- [x] The dir isn't a Terraform dir
- [x] Plan/JSON file does not exist
- [x] Terraform vars are missing
- [x] Terraform var file does not exist
- [x] Any error returned from Terraform binary
- [x] Any internal panic

### CODE REVIEW

File: internal/providers/terraform/aws/ecs_service.go [hunk 1]
```diff
@@ -46,6 +46,7 @@ func NewECSService(d *schema.ResourceData, u *schema.ResourceData) *schema.Resou
 		desiredCount = d.Get("desired_count").Int()
 	}
 	taskDefinition := d.References("task_definition")[0]
+	fmt.Println("HERE")
```
Comment (alikhajeh1) [inline internal/providers/terraform/aws/ecs_service.go:49]: oops

**Review by alikhajeh1 — APPROVED** [review]
LGTM, this is a huge usability improvement - thank you :)

File: internal/providers/terraform/provider.go [hunk 2]
```diff
@@ -39,103 +52,176 @@ func (p *terraformProvider) ProcessArgs(c *cli.Context) error {
...
-	if showDetails {
-		for rType, count := range unsupportedTypeCount {
-			message += fmt.Sprintf("\n%d x %s", count, rType)
+func terraformError(err error) {
+	if e, ok := err.(*TerraformCmdError); ok {
+		fmt.Fprintln(os.Stderr, indent(color.HiRedString("Terraform command failed with:"), "  "))
+		stderr := stripBlankLines(string(e.Stderr))
+		fmt.Fprintln(os.Stderr, indent(color.HiRedString(stderr), "    "))
+		if strings.HasPrefix(stderr, "Error: No value for required variable") {
+			fmt.Fprintln(os.Stderr, color.HiRedString("You can pass any Terraform args using the --tfflags option."))
```
Comment (alikhajeh1) [inline internal/providers/terraform/provider.go:207]:
```suggestion
			fmt.Fprintln(os.Stderr, color.HiRedString("Pass Terraform args using the --tfflags option."))
```

**Comment by aliscott** [comment]
Fixes https://github.com/infracost/infracost/issues/146

**Pull request merged**